### DISCUSSION TITLE
Feature Request: support hf:// scheme

### DISCUSSION THREAD

**Discussion by severo** [duckdb/duckdb#9691] [Ideas]
Parquet files stored on S3 can be accessed using s3://, see https://duckdb.org/docs/archive/0.9.2/guides/import/s3_import.

At Hugging Face, we created the hf:// scheme to access files stored on the Hub. See the Python implementation at https://huggingface.co/docs/huggingface_hub/v0.19.3/en/package_reference/hf_file_system.

We can use it as follows:

```python
from huggingface_hub import HfFileSystem
import duckdb
duckdb.register_filesystem(HfFileSystem())
duckdb.sql("SELECT SUM(size) FROM 'hf://datasets/vivym/midjourney-messages/data/*.parquet';")
```

Ideally, this scheme would be supported natively, without the need to register a filesystem, and not just for Python. For example, in the DuckDB CLI, we could run:

```sql
SELECT SUM(size) FROM 'hf://datasets/vivym/midjourney-messages/data/*.parquet';
```

**Comment by douenergy** [comment]
@severo really great work 👍

DuckDB also support other cloud blob prefix https://github.com/duckdb/duckdb/pull/9388 ( just get merged last week !)

* Cloudflare R2:// 
* GCS gcs://

**Comment by severo** [comment]
Closing since it has been implemented! Thanks to the team 🫶

(detail: it's still not included in duckdb wasm)
